src/hooks/useActionCenterSubscriber/useEvergreenDepositActionSubscriber.tsx

```diff
@@ -0,0 +1,85 @@
+import { usePrevious, useToast } from '@chakra-ui/react'
+import { fromAccountId } from '@shapeshiftoss/caip'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import { useEffect } from 'react'
+
+import { useWallet } from '../useWallet/useWallet'
+
+import { EvergreenDepositNotification } from '@/components/Layout/Header/ActionCenter/components/Notifications/EvergreenDepositNotification'
+import { actionSlice } from '@/state/slices/actionSlice/actionSlice'
+import { selectPendingEvergreenDepositActions } from '@/state/slices/actionSlice/selectors'
+import { ActionStatus } from '@/state/slices/actionSlice/types'
+import { selectTxs } from '@/state/slices/selectors'
+import { serializeTxIndex } from '@/state/slices/txHistorySlice/utils'
+import { useAppDispatch, useAppSelector } from '@/state/store'
+
+type UseEvergreenDepositActionSubscriberProps = {
+  onDrawerOpen: () => void
+  isDrawerOpen: boolean
+}
+
+export const useEvergreenDepositActionSubscriber = ({
+  onDrawerOpen,
+  isDrawerOpen,
+}: UseEvergreenDepositActionSubscriberProps) => {
+  const dispatch = useAppDispatch()
+  const {
+    state: { isConnected },
+  } = useWallet()
+
+  const toast = useToast({
+    duration: isDrawerOpen ? 5000 : null,
+    position: 'bottom-right',
+  })
+
+  const previousIsDrawerOpen = usePrevious(isDrawerOpen)
+
+  useEffect(() => {
+    if (isDrawerOpen && !previousIsDrawerOpen) {
+      toast.closeAll()
+    }
+  }, [isDrawerOpen, toast, previousIsDrawerOpen])
+
+  const pendingEvergreenActions = useAppSelector(selectPendingEvergreenDepositActions)
+  const txById = useAppSelector(selectTxs)
+
+  // Watch transaction status directly for each pending action
+  useEffect(() => {
+    if (!isConnected) return
+
+    pendingEvergreenActions.forEach(action => {
+      const { stakeTxHash, accountId } = action.evergreenDepositMetadata
+      if (!stakeTxHash || !accountId) return
+
+      const accountAddress = fromAccountId(accountId).account
+      const serializedTxIndex = serializeTxIndex(accountId, stakeTxHash, accountAddress)
+      const transaction = txById[serializedTxIndex]
+
+      if (
+        !transaction ||
+        (transaction.status !== TxStatus.Failed && transaction.status !== TxStatus.Confirmed)
```

Comment:
q: if we're going with handling all Tx statuses, do we even need this? Really only unknown should be a noop, and tbh I think we *should* be able to clean this one up from the app, that should never happen (i.e if we have a Tx, it's either pending, success or failed)